alf/algorithms/rl_algorithm.py

```diff
@@ -265,11 +366,47 @@ def predict(self, time_step: ActionTimeStep, state=None):
                 `action_distribution_spec`
               policy_step.state should be consistent with `predict_state_spec`
         """
-        policy_step = self.rollout(time_step, state)
+        if self.__class__.predict_action_distribution != RLAlgorithm.predict_action_distribution:
```

Comment:
perhaps we should find other way to do this

Reply:
removed